src/RedisRateLimiting/TokenBucket/RedisTokenBucketRateLimiter.cs

```diff
@@ -14,7 +15,12 @@ public class RedisTokenBucketRateLimiter<TKey> : RateLimiter
 
         private readonly TokenBucketLease FailedLease = new(isAcquired: false, null);
 
-        public override TimeSpan? IdleDuration => TimeSpan.Zero;
+        private int _activeRequestsCount;
+        private long _lastActiveTimestamp = Stopwatch.GetTimestamp();
+
+        public override TimeSpan? IdleDuration => Interlocked.CompareExchange(ref _activeRequestsCount, 0, 0) > 0
+            ? TimeSpan.Zero
+            : TimeSpan.FromTicks(Stopwatch.GetTimestamp() - _lastActiveTimestamp);
```

Comment:
Can we please change this to something similar to the runtime's [in-memory implementation](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/FixedWindowRateLimiter.cs#L36C9-L36C157)?

- Rename `_lastActiveTimestamp` to `_idleSince`
- Return `null` instead of `TimeSpan.Zero`
- Explicitly convert `Stopwatch` ticks into `TimeSpan` ticks using `TickFrequency`:
```C#
private static readonly double TickFrequency = (double)TimeSpan.TicksPerSecond / Stopwatch.Frequency;
public override TimeSpan? IdleDuration => _idleSince is null ? null : new TimeSpan((long)((Stopwatch.GetTimestamp() - _idleSince) * TickFrequency));
```
- Implement the same for the other rate limiters (fixed/sliding/concurrency)